lobsterpy/featurize/core.py

```diff
@@ -0,0 +1,829 @@
+        elif self.are_cobis:
+            cc, cw, cs, ck = self._calc_moment_features(label_list=label_list)
+            df.loc[ids, "bnd_wICOBI"] = per_bnd_xx
+            df.loc[ids, "antibnd_wICOBI"] = per_antibnd_xx
+            df.loc[ids, "w_ICOBI"] = w_icoxx
+            df.loc[ids, "EIN_ICOBI"] = ein_xx
+            df.loc[ids, "center_COBI"] = cc
+            df.loc[ids, "width_COBI"] = cw
+            df.loc[ids, "skewness_COBI"] = cs
+            df.loc[ids, "kurtosis_COBI"] = ck
+        else:
+            cc, cw, cs, ck = self._calc_moment_features(label_list=label_list)
+            df.loc[ids, "bnd_wICOHP"] = per_bnd_xx
+            df.loc[ids, "antibnd_wICOHP"] = per_antibnd_xx
+            df.loc[ids, "w_ICOHP"] = w_icoxx
+            df.loc[ids, "EIN_ICOHP"] = ein_xx
+            df.loc[ids, "center_COHP"] = cc
+            df.loc[ids, "width_COHP"] = cw
+            df.loc[ids, "skewness_COHP"] = cs
+            df.loc[ids, "kurtosis_COHP"] = ck
+
+        return df
+
+
+class FeaturizeCharges:
+    """
+    class to compute ionicity from CHARGE.lobster data
+
+    Args:
+        path_to_strucutre: path to POSCAR
+        path_to_charge : path to CHARGE.lobster (e.g., "CHARGE.lobster")
+        charge_type : set charge type used for computing ionicity. Possible options are "Mulliken" or "Loewdin"
+
+    Attributes:
+        get_df: pandas dataframe
+
+    """
+
+    def __init__(
+        self,
+        path_to_structure: str,
+        path_to_charge: str,
+        charge_type: str,
+    ):
+        self.path_to_structure = path_to_structure
+        self.path_to_charge = path_to_charge
+        self.charge_type = charge_type
+
+    def _calc_ionicity(self) -> float:
+        """
+        Method to calculate ionicity of crystal structure based on quantum chemical charges
+
+        I_("Charges ")=(1)/(N_("Atoms "))sum _(i)^(N_("Atoms "))((q_(i))/(v_("eff ",i)))
+
+        Returns:
+            Ionicity of the structure
+
+        """
+        chargeobj = Charge(filename=self.path_to_charge)
+        structure = Structure.from_file(self.path_to_structure)
+
+        if self.charge_type.lower() == "mulliken":
+            charges = chargeobj.Mulliken
+        elif self.charge_type.lower() == "loewdin":
+            charges = chargeobj.Loewdin
+        else:
+            raise ValueError(
+                "Please check the requested charge_type, "
+                'Possible options are "Mulliken" or "Loewdin"'
+            )
+
+        ch_veff = []
+        for i, j in enumerate(charges):
+            if (
+                j > 0
+                and not structure.species[i].is_transition_metal
+                and (
+                    not structure.species[i].is_actinoid
+                    and not structure.species[i].is_lanthanoid
+                )
+            ):
+                valence_elec = element(structure.species[i].value)
+                val = j / (valence_elec.nvalence() - 0)
+                ch_veff.append(val)
```

Comment:
This is not yet covered. You can also define a method/function for this.

Reply:
I realized, some redundant if-else checks were there, I simplified it now. Hope this should be fine. Am not able to figure out what you meant by having a function/method as  it is already part of method in the code.